NMS-15320: Added docs section for SameSite Cookie Attribute (#5902)
OpenNMS/opennms: docs/modules/operation/pages/deep-dive/admin/configuration/same-site-cookie.adoc

= SameSite Cookie Attribute

> The SameSite Cookie Attribute was introduced to control access to cookies and helps to prevent various XSS- and CSRF-attacks.

Reviewer: ```suggestion
The SameSite Cookie attribute controls access to cookies and helps prevent various cross-site scripting (XSS) and cross-site request forgery (CSRF) attacks.
```

This attribute tells browsers how to handle first- or third-party cookies and identifies whether to allow a cookie to be accessed.

> OpenNMS ships with the most secure configuration, *strict*, for its session cookie.

Reviewer 2: ```suggestion
{page-component-title} ships with the most secure configuration (`strict`) for its session cookie.
```

> This means that the session cookie will not be sent for any cross-site request.

Reviewer 2: ```suggestion
This means that the session cookie will not be sent for any cross-site requests.
```

> There may be environments where a user wants to relax this enforcement by setting the SameSite attribute to *lax*.

Reviewer 2: ```suggestion
There may be environments where a user wants to relax this enforcement by setting the SameSite attribute to `lax`.
```

> == Relaxing the SameSite Attribute enforcement policy

Reviewer: ```suggestion
== Relax the SameSite attribute enforcement policy
```

> == Relax the SameSite attribute enforcement policy

Reviewer 2: ```suggestion
== Relax enforcement policy
```

> In order to change the enforcement policy, you need to alter the file `$OPENNMS_HOME/jetty-webapps/opennms/WEB-INF/web.xml`.

Reviewer: ```suggestion
To change the enforcement policy, you need to alter the `$OPENNMS_HOME/jetty-webapps/opennms/WEB-INF/web.xml` file.
```

> To change the enforcement policy, you need to alter the `$OPENNMS_HOME/jetty-webapps/opennms/WEB-INF/web.xml` file.

Reviewer 2: ```suggestion
To change the enforcement policy, you must alter `$OPENNMS_HOME/jetty-webapps/opennms/WEB-INF/web.xml`:
```

> 

Reviewer 2: ```suggestion
+
```

[source, xml]
----
<session-config>
  <cookie-config>
    <http-only>true</http-only>

>     <comment>__SAME_SITE_STRICT__</comment> <1>

Author: Yes, the double undescores are needed.
